src/Physics/babylon.physicsImpostor.ts

```diff
@@ -330,13 +330,22 @@ module BABYLON {
             }
         }
 
+        /**
+         * Legacy collision detection event support
+         */
+        public onCollideEvent: (collider:BABYLON.PhysicsImpostor, collidedWith:BABYLON.PhysicsImpostor) => void = null;
+
         //event and body object due to cannon's event-based architecture.
         public onCollide = (e: { body: any }) => {
-            if (!this._onPhysicsCollideCallbacks.length) return;

```

Comment:
But i do the checking in the actual function that is assigned onCollideEvent...

Even without my code change, in the original code... if you pass an imposter, it would check (several times per frame) if the imposter is in the list of callbacks... foreach one returned it would invoke the callback... I am doing the same exact thing in the onCollideEvent (except i am checkk for tags in a list on the metadata)... The ONLY difference with my code... it actually THUNKS into another function (whatever is assign onCollideEvent) to do my checking... So i ignore everything that is not tagged... If is only mesh (imposter) 1 tagged then it will only call is handler function... WHICH IS THE SAME THING... Now the if (!this._onPhysicsCollideCallbacks.length) return; get called right after that... if no other callbacks registered it will bail out... The mod code just gives me a chance to handle the RAW collision checking MYSELF... Should i choose to... I am just as carefule in my processCollisionEventHandler:


Here is the code that actually executes for onCollideEvent:

`        private processPhysicsCollisions(collider: PhysicsImpostor, collidedAgainst:PhysicsImpostor) : void {
            if (collider.object != null && collidedAgainst.object != null) {
                var colliderAny:any = collider.object;
                if (colliderAny != null && colliderAny.metadata != null && colliderAny.metadata.collisionTags != null && colliderAny.metadata.collisionEvent != null) {
                    var collidedAgainstAny:any = collidedAgainst.object;
                    if ( collidedAgainstAny != null && collidedAgainstAny.metadata != null && collidedAgainstAny.metadata.tagName != null) {
                        if (colliderAny.metadata.collisionTags.length === 0 || colliderAny.metadata.collisionTags.indexOf(collidedAgainstAny.metadata.tagName) !== -1) {
                            colliderAny.metadata.collisionEvent(collidedAgainstAny);
                        }
                    }
                }
            }
        }
` 
I just dont see the problem... But thats kool... its your code... I would NOT want folks just mucking around with my shit either ... I will figure out something else for my toolkit :)

As far as the particle System... I think i just changed different files to fix different issue... It would not allow me to cerate a seperate pull request... it just kept putting my changes onto the current open pull request... I dont know what hell i am doing most of the time with the github syncing... Most if the time i WAIT till one PR goes all the way thru then make a new one for next issue (i dont know why i didnt wait til you made a decision before adding the particle system changes... SORRY )

Do i have to take out all 3 of these commits then???